metadata/src/main/java/org/apache/kafka/controller/FeatureControlManager.java

```diff
@@ -344,7 +344,7 @@ private ApiError updateMetadataVersion(
                 log.warn("Downgrading metadata.version from {} to {}.", currentVersion, newVersion);
             } else if (allowUnsafeDowngrade) {
                 return invalidMetadataVersion(newVersionLevel, "Unsafe metadata downgrade is not supported " +
-                        "in this version.");
+                        "in this Kafka version.");
```

Comment:
1. I'm thinking we should also change the error message in L350-352, to not mention anything about `UNSAFE_DOWNGRADE`. Also add a comment above L350 saying we should add it back once unsafe downgrade is implemented.
2. `Unsafe metadata downgrade is not supported in this Kafka version` will make users wonder "in which kafka version" it will be supported. Maybe we can just say: `Unsafe metadata downgrade is not supported. ` WDYT?

Also, we should add a sub-task in [KAFKA-14175](https://issues.apache.org/jira/browse/KAFKA-14175) for document update after unsafe downgrade is implemented.

Reply:
Agree with both points. Should we log another Jira linked to KAFKA-13410 for the missing functionality, which also includes updating the documentation?

Reply:
Sounds good to me.